src/Covenant/ASG.hs

```diff
@@ -185,12 +199,16 @@ bound Scope = Bound . fromIntegral . natVal $ Proxy @n
 -- @since 1.0.0
 lam ::
   forall (n :: Natural) (m :: Natural).
+  ASGType ->
```

Comment:
We need to update the documentation to specify that this is the intended type of the argument, not the type of the lambda as a whole.